src/Orleans.Core/Networking/Connection.cs

```diff
@@ -196,9 +199,10 @@ private void CloseInternal(Exception exception)
 
                     try
                     {
-                        if (_processOutgoingTask is Task task && !task.IsCompleted)
+                        if (!outgoingCompleted)
                         {
                             this.Context.Transport.Output.CancelPendingFlush();
+                            outgoingCompleted = true;
                         }
```

Comment:
I think the result here is that it ends up not gracefully draining the connection before shutdown